### PR TITLE
Create scaffold for "Overlays" on a users screen & add User Info Overlays

### PR DESCRIPTION
**Pull request by GailWilson**
## Summary
Within this PR we add the scaffolding needed to have overlays on a specified user's video screen.

Details in notion doc: https://www.notion.so/Overlays-b9bc5d22e01645ba8394fbba0655ce67
Image of where we want each overlay to go:
Whiteboard 3 -01

Name & icons along the top bar:
Screen Shot 2020-03-29 at 7 02 30 PM

Scrollability when many things are added to the right portion:
Screen Recording 2020-03-29 at 05 08 PM

- Commit 1 : Carlos added the quick scaffolding
- Commit 2 : Gail renames/organizes folders/files to make room for the overlays
- Commit 3 : Gail sets up the registry/context/provider/components for overlays & User Info overlays
- Commit 4 : Gail updates specs

Within this specific PR we are:
- [x] Adding scaffolding to get the overlays for a participant
- [x] Adding the overlay for the UserInfo bar at the top of the screen

## Tests
- [x] Test locally
- [x]  Update & add specs

### CODE REVIEW

File: src/Overlay.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { FunctionComponent } from 'react';
+
+export interface UserInfoOverlayType {
```
Comment (carlosdp) [inline src/Overlay.ts]: Can we rename these to `Definition`, ie. UseInfoOverlayDefinition

File: src/components/ParticipantScreen/index.tsx [hunk 2]
```diff
@@ -109,20 +88,9 @@ export default function ParticipantInfo({
...
-        </div>
-        <div>
-          {!isAudioEnabled && <MicOff data-cy-audio-mute-icon />}
-          {!isVideoEnabled && <VideocamOff />}
-          {isScreenShareEnabled && <ScreenShare />}
-          {isSelected && <PinIcon />}
-        </div>
+        <UserInfoOverlayArea participant={participant} overlays={overlays?.userInfoOverlays} />
       </div>
+      {/* TODO(gail.wilson) -- Make Bandwidth warning a "screen takeover" overlay of some sort */}
```
Comment (carlosdp) [inline src/components/ParticipantScreen/index.tsx]: I think the Bandwidth Warning should maybe be moved out of here and into a child component that wraps the video stream directly. That way it's always underneath the overlays.

Comment (carlosdp) [reply src/components/ParticipantScreen/index.tsx]: We don't have to do this now though.

Comment (GailWilson) [reply src/components/ParticipantScreen/index.tsx]: Cool I'll leave the comment so we can make sure to address at some point

**Pull request merged**